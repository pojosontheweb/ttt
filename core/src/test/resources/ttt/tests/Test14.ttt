<%@ page import="com.xyz.myapp.Foo" %>
<%@ page extends="java.io.Serializable" %>
<%!
    /* multi line comments are skipped */

    Foo foo;

    /*
        real multi !
     */
    int bar;

    /* no shit
     */
%>
<div class="foo">
    <span><%= foo.getBar() %></span>
    <ul>
    <%
        // comment in script block
        for(int i = 0; i < bar; i++) { /* multi comment in script block */
    %>
        <li>i equals <%= i // in scriptlet %> </li>
    <%
        }
        /* multi
        on several
        rows
         */
    %>
    </ul>
</div>