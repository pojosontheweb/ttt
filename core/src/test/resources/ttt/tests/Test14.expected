package ttt.tests;

import com.xyz.myapp.Foo;

public class Test14 extends com.pojosontheweb.ttt.Template implements java.io.Serializable {

	private final Foo foo;
	private final int bar;

	/**
	 * Creates an instance of this template.
	 *
	 */
	public Test14(Foo foo, int bar) {
		this.foo = foo;
		this.bar = bar;
	}

	@Override
	public void render(java.io.Writer out) throws java.io.IOException {
		write(out, "<div class=\"foo\">\n    <span>" );
		write(out,  foo.getBar()  );
		write(out, "</span>\n    <ul>\n    " );
		        // comment in script block
        for(int i = 0; i < bar; i++) { /* multi comment in script block */
    
		write(out, "\n        <li>i equals " );
		write(out,  i // in scriptlet  );
		write(out, " </li>\n    " );
		        }
        /* multi
        on several
        rows
         */
    
		write(out, "\n    </ul>\n</div>" );
	}
}
